account_billing/models/account_billing.py

```diff
@@ -0,0 +1,258 @@
+        string='Company',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.billing'),
+        index=True,
+        help='Leave this field empty if this route is shared \
+            between all companies',
+    )
+    date = fields.Date(
+        string='Billing Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=fields.Date.context_today,
+        help='Effective date for accounting entries',
+        track_visibility='always',
+    )
+    due_date = fields.Date(
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        string='Due Date Threshold',
+        default=lambda self: fields.Date.context_today(self),
+        required=True,
+        track_visibility='always',
+    )
+    invoice_ids = fields.Many2many(
+        comodel_name='account.invoice',
+        relation='account_billing_invoice_rel',
+        column1='billing_id',
+        column2='invoice_id',
+        string='Invoices',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self._context.get('active_ids', []),
+    )
+    invoice_related_count = fields.Integer(
+        string='# of Invoices',
+        compute='_compute_invoice_related_count',
+        help='Count invoice in billing',
+    )
+    state = fields.Selection(
+        selection=[
+            ('draft', 'Draft'),
+            ('cancel', 'Cancelled'),
+            ('billed', 'Billed'),
+        ],
+        string='Status',
+        readonly=True,
+        default='draft',
+        help='''
+            * The 'Draft' status is used when a user create a new billing\n
+            * The 'Billed' status is used when user confirmed billing,
+                billing number is generated\n
+            * The 'Cancelled' status is used when user billing is cancelled
+        ''',
+    )
+    narration = fields.Text(
+        string='Notes',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='Notes',
+    )
+    bill_type = fields.Selection(
+        selection=[
+            ('out_invoice', 'Customer Invoice'),
+            ('in_invoice', 'Vendor Bill'),
+        ],
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self._context.get('bill_type', False),
+        help='Type of invoice',
+    )
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        string='Currency',
+        required=True,
+        default=lambda self: self._get_currency_id(),
+        help='Currency',
+    )
+    billing_line_ids = fields.One2many(
+        comodel_name='account.billing.line',
+        inverse_name='billing_id',
+        string='Bill Lines',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.onchange('partner_id')
+    def _onchange_invoice_id(self):
+        inv = self.env['account.invoice'].browse(
```

Comment:
Name it with 's', to signify multi
```suggestion
        invoices = self.env['account.invoice'].browse(
```